Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/ComponentEditor.cpp

```diff
@@ -214,8 +211,34 @@ namespace AzToolsFramework
             if (!aggregateInstance)
             {
                 // Set the adapter component to this instance.
-                // Note: multiple selection with DPE is not yet supported
                 m_adapter->SetComponent(componentInstance);
+
+                // set the DPE back to the regular adapter and destroy the aggregate adapter
+                if (m_aggregateAdapter)
+                {
+                    m_dpe->SetAdapter(m_adapter);
+                    m_aggregateAdapter.reset();
+                }
+            }
+            else
+            {
+                if (!m_aggregateAdapter)
+                {
+                    m_aggregateAdapter = AZStd::make_shared<AZ::DocumentPropertyEditor::LabeledRowAggregateAdapter>();
+
+                    // for now, disable "values differ rows", since there are so many pointer and opaque types in the Inspector
+                    // and the output is noisy and unpleasant.
+                    m_aggregateAdapter->SetGenerateDiffRows(false);
+                    m_aggregateAdapter->AddAdapter(m_adapter);
+                    m_filterAdapter->SetSourceAdapter(m_aggregateAdapter);
+                    m_dpe->SetAdapter(m_filterAdapter);
+                }
+
+                AZStd::shared_ptr<AZ::DocumentPropertyEditor::ComponentAdapter> newAdapter =
+                    (Prefab::IsInspectorOverrideManagementEnabled() ? AZStd::make_shared<Prefab::PrefabComponentAdapter>()
+                                                                    : AZStd::make_shared<AZ::DocumentPropertyEditor::ComponentAdapter>());
+                newAdapter->SetComponent(componentInstance);
+                m_aggregateAdapter->AddAdapter(newAdapter);
```

Comment:
Doesn't this mean that when AddInstance() is called on the aggregate instance, we will be adding 2 adapters instead of 1 for the aggregate adapter? Is that by design, if so, a comment here would be nice.

Reply:
Yep, I'm happy to comment it, but to answer your question... If you look at EntityPropertyEditor::BuildSharedComponentUI, you'll see the loop that begins:

for (AZ::Component* componentInstance : componentInstances)

the first iteration always has a null aggregateInstance, due to  this line:

AZ::Component* aggregateInstance = componentInstance != componentInstances.front() ? componentInstances.front() : nullptr;

so in the multi-edit code, if there's no aggregateInstance specified (it's null because it's the first, and possibly only, component) we create the right componentAdapter and set it up just as before. If there are subsequent iterations of that loop, it means we're in multi-edit, and aggregateInstance is filled in with an additional component. We then make an AggregateAdapter if it doesn't exist yet, make a normal component adapter for this new instance, then add it to the AggregateAdapter.

If you're wondering if it's adding duplicate adapters, it isn't.

Reply:
added comments, switched to adapter factory function so only the Inspector attempts to use the DPE